action.yml

```diff
@@ -68,26 +80,59 @@ runs:
       shell: bash
 
     - name: Fetch report from ${{ inputs.diff-storage }}
-      if: contains(inputs.diff, 'true')
       uses: actions/checkout@v2
       with:
         path: ${{ inputs.diff-storage }}
         fetch-depth: 0
 
+    - name: Initialize storage branch
+      working-directory: ${{ inputs.diff-storage }}
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Actions"
+        # Switch to the branch if it already exists
+        git switch ${{ inputs.diff-storage }} || true
+        git pull origin ${{ inputs.diff-storage }} || true
+        # Create the branch if it doesn't exist yet
+        git checkout --orphan ${{ inputs.diff-storage }} || true
+        # Commands below will have no effect on the storage branch that already exists.
+        # No changes will be committed.
+        # Prepare the orphan branch containing only SVG badges and coverage report for diff-branch.
+        # Iterate through directories in top-level directory in repo.
+        # Remove anything, except .git, badges and coverage-report-action.
+        find . -mindepth 1 -maxdepth 1 ! \( -name .git -o -name badges -o -name coverage-report-action \) -type d -exec rm -rf {} +
+        # Remove other files in top-level directory.
+        find . -mindepth 1 -maxdepth 1 -type f -delete
+        # Initial commit has to contain some file
+        touch README.md
+        git add --all -f
+        git commit -m "Initialize storage branch" || true
```

Comment:
In the near future, we should make this its own Github Action. For now, this looks great.

Reply:
Okay.